sycl/test/built-ins/printf.cpp

```diff
@@ -0,0 +1,121 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out | FileCheck %s
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+#include <CL/sycl.hpp>
+
+#include <cstdint>
+
+using namespace cl::sycl;
+
+// According to OpenCL C spec, the format string must be in constant address
+// space
+#ifdef __SYCL_DEVICE_ONLY__
+#define CONSTANT __attribute__((opencl_constant))
+#else
+#define CONSTANT
+#endif
+
+
+// This is one of the possible ways to define a format string in a correct
+// address space
+static const CONSTANT char format_hello_world[] = "Hello, World!\n";
+
+// Static isn't really needed if you define it in global scope
+const CONSTANT char format_int[] = "%d\n";
+
+static const CONSTANT char format_vec[] = "%d,%d,%d,%d\n";
+
+const CONSTANT char format_hello_world_2[] = "%lu: Hello, World!\n";
+
+int main() {
+  {
+    default_selector Selector;
+    queue Queue(Selector);
+
+    Queue.submit([&](handler &CGH) {
+      CGH.single_task<class integral>([=]() {
+        // String
+        intel::experimental::printf(format_hello_world);
+        // CHECK: Hello, World!
```

Comment:
CI failure on Windows seems to me some Windows-specific problem rather than the patch issue:

When I launch this app, I see the following output:
```
Hello, World!
123
-123
33.400002
-33.400002
5,6,7,8
8,7,6,5
00000057519EDF34
0: Hello, World!
5: Hello, World!
3: Hello, World!
4: Hello, World!
1: Hello, World!
2: Hello, World!
6: Hello, World!
7: Hello, World!
8: Hello, World!
9: Hello, World!
```

Which seems to be correct. However, if I redirect the output into a file or into the input to another tool (FileCheck, for example), I see the following order of statements:
```
123
-123
33.400002
-33.400002
5,6,7,8
8,7,6,5
0000008C57F0DF54
0: Hello, World!
5: Hello, World!
7: Hello, World!
8: Hello, World!
6: Hello, World!
9: Hello, World!
1: Hello, World!
2: Hello, World!
3: Hello, World!
4: Hello, World!
Hello, World!
```

So, the very first `Hello, World` has moved to the end of the output and broke `CHECK`-lines

Right now I cannot really explain this. @asavonic, any ideas?

Reply:
This seems to be a known issue of Intel CPU Runtime for OpenCL. I will `xfail` this test on Windows for now